workflow/controller/controller.go

```diff
@@ -629,8 +629,11 @@ func (wfc *WorkflowController) processNextItem(ctx context.Context) bool {
 			}
 		}
 		if doPodGC {
-			for podName := range woc.completedPods {
-				woc.controller.queuePodForCleanup(woc.wf.Namespace, podName, deletePod)
+			labelSelector := woc.execWf.Spec.PodGC.GetLabelSelector()
+			for podName, podSummary := range woc.completedPods {
+				if labelSelector == nil || podSummary.matched {
```

Comment:
I about to pivot - I think we should only add items to completedPods if they match

Reply:
you can change completedPods to `map[string]corev1.PodPhase`

Reply:
Great idea. The code looks much cleaner now. Thanks!